app/services/users_service.rb

```diff
@@ -41,6 +34,24 @@ def register(email, password, organization_name)
     result
   end
 
+  def register_from_invite(email, password, organization_id)
+    result.user = User.find_or_initialize_by(email: email)
+
+    if result.user.id
+      result.fail!(code: 'user_already_exists')
+
+      return result
+    end
+
+    ActiveRecord::Base.transaction do
+      result.organization = Organization.find(organization_id)
+      result.token = generate_token
+      create_user_and_membership(result, result.organization, password)
```

Comment:
you already have `organization` inside `result` :)

Reply:
What do you mean ? this endpoint is called while the user is not connected so I need to fetch the orga by id

Reply:
Oh you mean it's not necessary to pass it as argument ?

Reply:
removed